sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JsonInferSchema.scala

```diff
@@ -68,14 +83,18 @@ private[sql] class JsonInferSchema(options: JSONOptions) extends Serializable {
             Some(inferField(parser))
           }
         } catch {
-          case  e @ (_: RuntimeException | _: JsonProcessingException) => parseMode match {
-            case PermissiveMode =>
-              Some(StructType(Seq(StructField(columnNameOfCorruptRecord, StringType))))
-            case DropMalformedMode =>
-              None
-            case FailFastMode =>
-              throw QueryExecutionErrors.malformedRecordsDetectedInSchemaInferenceError(e)
-          }
+          case e: SparkUpgradeException => throw e
+          case e @ (_: RuntimeException | _: JsonProcessingException |
+                    _: MalformedInputException) =>
```

Comment:
Could you write tests or modify existing tests to check new exception handling:
- SparkUpgradeException
- MalformedInputException

Reply:
Good point.  Will add later.

Reply:
The `SparkUpgradeException` is not necessary, so I removed this exception handling.
I add a new test to cover `MalformedInputException`